tests/Console/Output/ProcessOutputFactoryTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Console\Output;
+
+use PhpCsFixer\Console\Output\NullOutput;
+use PhpCsFixer\Console\Output\OutputContext;
+use PhpCsFixer\Console\Output\ProcessOutputFactory;
+use PhpCsFixer\Console\Output\ProcessOutput;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Output\NullOutput as SymfonyNullOutput;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Console\Output\ProcessOutputFactory
+ */
+final class ProcessOutputFactoryTest extends TestCase
+{
+    /**
+     * @dataProvider provideValidProcessOutputContexts
+     */
+    public function testValidProcessOutputIsCreated(
+        string $outputType,
+        OutputContext $context,
+        string $expectedOutputClass
+    ): void {
+        static::assertInstanceOf($expectedOutputClass, ProcessOutputFactory::create($outputType, $context));
+    }
+
+    public static function provideValidProcessOutputContexts(): iterable
+    {
+        $context = new OutputContext(new SymfonyNullOutput(), new EventDispatcher(), 100, 10);
+        $nullContext = new OutputContext(null, new EventDispatcher(), 100, 10);
+
+        yield 'none' => ['none', $context, NullOutput::class];
+
+        yield 'dots' => ['dots', $context, ProcessOutput::class];
```

Comment:
why not using `OUTPUT_TYPE_DOTS` ?

Reply:
You mean something like this?

```php
        yield 'none' => [ProgressOutputType::NONE, $context, NullOutput::class];

        yield 'dots' => [ProgressOutputType::DOTS, $context, ProcessOutput::class];
```

I wasn't sure if this should be dynamic thing or explicit value. Like in migrations, you shouldn't use dynamic content from the code, rather explicit values representing state at particular point in time, so they don't break when e.g. constant is changed. If actual values (constants) are changed then test fails and requires manual action which is rather good - otherwise it could still work but could make no sense, like:

```php
        yield 'dots' => [ProgressOutputType::SOME_RENAMED_CONST, $context, ProcessOutput::class];
```

Key does not correspond with progress type here. But it's not strong opinion, I don't have established standard for that, so only thinking out loud 😉.

Reply:
If someone would replace value under constant, I wouldn't like to need to update all tests to reflect new value. That's exactly why we have the constant - to not use value itself, again and again, and then manually update everywhere.

about the yielding key, I'm actually OK to have it follow the constant NAME rather than it's value.

Reply:
> That's exactly why we have the constant - to not use value itself, again and again, and then manually update everywhere.

I agree, in terms of the production code. Not sure about tests 🤷‍♂️. But OK, I'll change it 👍